### DISCUSSION TITLE
SET_SHORTCUTS error

### DISCUSSION THREAD

**Discussion by marcin-brumer** [johnlindquist/kit#1450] [Errors]
Hello, I get this error when opening kit prompt. It doesn't happen every time, but often enough to be very bothersome. Any ideas and/or suggestion would be much appreciated. 

```
Timeout after 1 seconds waiting for SET_SHORTCUTS response       
The app failed to send a SET_SHORTCUTS response to the script process within the expected timeframe. Halting script.

**Comment by asos-danielc** [comment]
I too am getting this, however quite a lot

```
Timeout after 1 seconds waiting for SET_SHORTCUTS response
              
The app failed to send a SET_SHORTCUTS response to the script process within the expected timeframe. Halting script.

Please share this error to our GitHub Discussions with your scenario: https://github.com/johnlindquist/kit/discussions/categories/errors

Error: Timeout after 1 seconds waiting for SET_SHORTCUTS response
              
The app failed to send a SET_SHORTCUTS response to the script process within the expected timeframe. Halting script.

Please share this error to our GitHub Discussions with your scenario: https://github.com/johnlindquist/kit/discussions/categories/errors

    at Timeout._onTimeout (file:///Users/DanielC/.kit/target/app.js:1605:25)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
```

**Reply by johnlindquist** [reply]
@marcin-brumer @asos-danielc  (sorry for missing the original post)

Can you try updating:

`~/.kit/target/app.js`

Search for:

```
await sendWait(Channel.SET_SHORTCUTS, shortcuts)
```

Change it to:

```
send(Channel.SET_SHORTCUTS, shortcuts)
```

I'll hunt down the core of the issue. Does this happen on a specific script? Or from the main menu?

**Reply by asos-danielc** [reply]
I will give a go as soon as I can... away from keyboard atm. Thanks for the reply. It's happening for me on the main script

**Reply by asos-danielc** [reply]
Do I need to rebuild kit or reload it in a particular way?

**Reply by johnlindquist** [reply]
@asos-danielc No, the App loads a fresh SDK with each new Window. (although it might be cached the first time)

You can add `console.log('hi')` statements to make sure it's picking it up

**Reply by asos-danielc** [reply]
Just tested, same error unfortunately

**Reply by johnlindquist** [reply]
@asos-danielc Just to confirm the app.js is properly reloading, does commenting out this prevent it?

app.js
```
          rej(
            new Error(
              `Timeout after ${
                timeout / 1000
              } seconds waiting for ${channel} response
              
The app failed to send a ${channel} response to the script process within the expected timeframe. Halting script.

Please share this error to our GitHub Discussions with your scenario: https://github.com/johnlindquist/kit/discussions/categories/errors
`,
              {
                cause: "timeout",
              }
            )
          )
          ```

**Reply by asos-danielc** [reply]
The app reloads with the changes (hi shown), commenting out the above keeps the app loaded but does show the timeout error sporadically still

**Reply by johnlindquist** [reply]
@asos-danielc to be clear, it's still the "SET_SHORTCUTS" timeout version of the error?

**Reply by asos-danielc** [reply]
Typically it isnt happening at the moment, but I am reasonably confident it was that. Apologies on the delay

**Reply by marcin-brumer** [reply]
Hello @johnlindquist, thx for your answer, I have used `Force Repair SDK` few days ago, which fixed the issue for me and didn't used script kit much since then but I assume that problem will be back. I did full reinstall of the app few weeks ago and it only fixed problem for day or two, so I assume that `Force Repair SDK` will be no different, but hey, maybe I am wrong 😄 I will get back to you if the error will happen again.

**Reply by marcin-brumer** [reply]
OK @johnlindquist just got the same error again, tried `send(Channel.SET_SHORTCUTS, shortcuts)` but it didn't help, still get `SET_SHORTCUTS` error. App did reloaded because I can see consol.log added by me. Anything that I can provide to help find root case of the problem?
